src/main/java/com/google/api/codegen/CommentPatterns.java

```diff
@@ -26,6 +26,7 @@
       Pattern.compile("\\[([^\\]]+)\\]\\((\\p{Alpha}+:[^\\)]+)\\)");
   public static final Pattern CLOUD_LINK_PATTERN =
       Pattern.compile("\\[([^\\]]+)\\]\\(((?!\\p{Alpha}+:)[^\\)]+)\\)");
-  public static final Pattern PROTO_LINK_PATTERN = Pattern.compile("\\[([^\\]]+)\\]\\[[^\\]]*\\]");
+  public static final Pattern PROTO_LINK_PATTERN =
+      Pattern.compile("\\[([^\\]]+)\\]\\[([A-Za-z_][A-Za-z_.0-9]*)*\\]");
```

Comment:
Actually there still be problem with the current approach. For example, [A-Za-z_$][a.b.c.d] is considered to be valid proto link, since we allow '$' within the first pair of [] , so a.b.c.d will be removed. So as long as comment string contains [<anything>][a.b.c.d], it will be parsed as a valid proto link, which will have the problem of removing the content within the second pair of bracket. As @jmuk mentioned, #331 may be the right approach. But in the mean time, we will have to live with is imperfect solution.